### DISCUSSION TITLE
Domain for Sympa project

### DISCUSSION THREAD

**Discussion by ikedas** [sympa-community/sympa#1625] [Team Posts]
Currently the site [www.sympa.org](https://www.sympa.org) (and the domain `sympa.org`) is not under control of our project. I think it is unlikely that these will be under our control in (at least) near future. 

OTOH currently, GitHub provides most of the functionality needed for Sympa development (the individual platform packagers are working in their own environments). I'd point out the following immediate problems:
- The documentation site [sympa-community.github.io](https://sympa-community.github.io) uses a bit long URL. 
- The [translation site](https://translate.sympa.org/) is external and is due for replacement soon, i.e. its IP must be changed.

The common challenge is that we don't have a domain name (and name servers) for our project.
I have currently reserved a domain name `sympa.community` and am willing to offer it to the Sympa community.  If it could be used, the problems above can be solved:

- The documentation site may be renamed to shorter name using the new domain (See [GitHub Docs documentation](https://docs.github.com/pages/configuring-a-custom-domain-for-your-github-pages-site) for technical details).
- When the translation site will be replaced, we no longer worry about managing `sympa.org` domain: We may use the new domain instead.

I'd like to hear your thoughts on having our own domain name (and DNS).

---
N.B.: Another issue is what to do with community-related mailing lists which are hosted by RENATER now, but it won't be covered in this thread.

**Comment by ikedas** [comment]
I had proposed to my company to host the new domain and the new VM for translation server, and they have asked me to submit a request for approval regarding it. So, we will probably be able to register the new domain name in the DNS by the end of June.

As I wrote, the most pressing issue is the replacement of the translation server. We have a chance to solve at least it.

### Request for help

If anyone is willing to host one of the DNS servers for the new domain, please come forward.

**Comment by ikedas** [comment]
I have submitted a request to my company and it has been approved.  Below is translated content of that request.

---
## Request for Approval (CONFIDENTIAL)


From: IKEDA Soji, System solution group #1, IT solution dept.
Date: 2022-06-06

**Subject: Equipment for Sympa Development Community**

Contents

* Purpose

  To support the activities of the Sympa development community.

* Background

  Sympa, an open source software, is currently being developed by volunteers in an online community centered on GitHub.  The significance of this activity has been recognized by us, and Ikeda and others within us have been allowed to participate in the development community.  In addition, they have received material assistance in the form of equipment for their activities. We are now discussing the continuation of this assistance in order to upgrade the facilities.

* Contents

  1. A set of server machines for Sympa development: 23,738 yen/year [Ref. 1]
       - Rights for usage of VPS control panel by us are included.
       - The "Sympa Japanese Information" server, which was previously hosted by us, is no longer active and will be discontinued.
  2. Domain name registration and DNS hosting for the community: 4,598 yen/year [Ref. 2]
       - Due to unsuccessful negotiations on the transfer of the official domain name "sympa.org" to the community, a new domain name (already acquired) will be hosted.
       - DNS hosting is included.
       - The domain name "sympa-ja.org", which was previously hosted by us, will be discontinued as it is no longer active.

* Other points to note

  The purpose of this request is to support the community and not to reflect our company's intentions in the community's management policy.

**Comment by racke** [comment]
Excellent :+1: Kudos for your company :star_struck:

**Comment by ikedas** [comment]
## Migration of lists [proposal]

Hi @everyone,

DNS for the new domain is expected to be up by the end of this week. We will also have to migrate the mailing lists about Sympa to the new domain in near future.  Below is the proposal.

* First, I think we'd like to migrate the `devel` and `security` mailing lists which are public contacts of the development of Sympa.  For me, the migration will be ready by early July at the latest.

* An announcement of the migration must be made. If we can announce it in time for the next release, we would like to do so. @racke, do you think when is it likely to happen?

Please give any input on the migration of the other lists as well.

| Current list (`@listes.renater.fr`, uoi)  | Migrated to (`@sympa.community`) [proposed] | comments |
| - | - | - |
| `sympa-announce`      | `announce`          | The list including `-dev`, `-developpers`, `-es`, `-fr`, `-packagers` & `-users` below. |
| `sympa-authors` | (won't be migrated) | Already out of use. Contact information for the author has been taken over by `-developpers` below. |
| `sympa-commits` | (won't be migrated) | The list tracking commits on subversion repository (We possiblly may need similar one for git in the future). |
| `sympa-dev`          | `devel` (see note below)  | Out of use (1999-2010). Taken over by `-developpers` below. |
| `sympa-developpers`  | `devel` (see note below) | Public contact for the authors. |
| `sympa-es`           | `es`                | No traffic yet. |
| `sympa-fr`           | `fr`                | |
| `sympa-packagers`    | `packagers`         | |
| `sympa-security`     | `security`          | Contact about security incident. |
| `sympa-users`        | `en` (_Is this renaming appropriate?_) | |
| `lets@translate.sympa.org` | `translate`   | The list including users of translation server.  This should be included in the new `-announce` list above. |
| `users@sympa-ja.org` | `ja` | Dormant. However, I'd like to transfer this to the community at this time. |

Notes:

  * On `sympa-dev` & `sympa-developpers`, only archives should be migrated to the new list.  Subscribers for the new list `devel` will be newly decided --- for the time being, would be the memebers of this team?.

  * RENATER members have served as admins of the lists above. Replacements will have to be chosen, but I may take over for the time being.   Exceptions are `-es`, `ja` & `security`: These have own admins already. For these lists, it would be better to migrate the current admins.

---
EDIT: Some typos & corrections. Added `sympa-commits`.
EDIT: Change proposed name `dev` to `devel`.

**Comment by racke** [comment]
I'm very busy with a project till end of June / beginning of July. After that I will look at the release process for 6.2.70.
Who will be able to administer the Sympa instance itself? We learned that a single entity managing a project can cause problems ...

**Comment by ikedas** [comment]
> I'm very busy with a project till end of June / beginning of July. After that I will look at the release process for 6.2.70.
> Who will be able to administer the Sympa instance itself? We learned that a single entity managing a project can cause problems ...

I will.  From the beginning I already have been managing a mailing list server for Sympa project.

I am already administering a mailing list server for a mailing list lets@translate.sympa.org.  This time, it will be migrated to the server with new domain (see previous my post).

As you know, managing a mailing list server, there are no difference either it has only one list or ten.

- Of course I’m not alone.  In worst case, the colleagues of my company may support administration of the server.  --- I'm not so inexperienced on the system managemrnt as to make these proposal without a lifeline in place.😤

**Comment by ikedas** [comment]
Also, I can explain the advantage on managing our own mailing list server. Ill do it tomorrow (such job 
 is pain for me with mobile phone...).

**Comment by racke** [comment]
Yeah, I have no doubt you can look after a Sympa server. Just making sure that emergencies are covered.

**Comment by ikedas** [comment]
It may possibly be possible to install ssh public key of you (if you like) and/or someone. If the need to do so arises in the future, we can consider it then.

Besides, we may expect 6.2.70 release process to start in July.  Migration of the lists could be done accordingly at that time.

  * Only `translate` list will be migrated first. This is tied to the translation server and is announcement-only list.

**Comment by ikedas** [comment]
I'm considering the following as DNS RRs for the new domain. Please point out any omissions.

Most of records refers to the new server (except hosted NS and www).  Both IPv4 & IPv6 addresses are used but latter may be removed because it is not necessary.

``` zone
$ORIGIN sympa.community.
$TTL 86400

@          IN SOA   (...)
           IN NS    [hosted name server]
           IN NS    [hosted name server]
           IN NS    [hosted name server]
           IN NS    [hosted name server]

           IN MX    10 mx-v4            ; SMTP: MXs for list server on its own
           IN MX    10 mx-v6
           IN TXT   "v=spf1 +a +mx ~all" ; SPF: allow those from list server itself
           IN A     [IPv4 address]      ; HTTP:  apex domain redirected to www

mx-v4      IN A     [IPv4 address]
mx-v6      IN AAAA  [IPv6 address]
_domainkey IN NS    mx-v4               ; DKIM/ARC: delegate key management to list server
           IN NS    mx-v6

lists      IN A     [IPv4 address]      ; HTTP: WWSympa
           IN AAAA  [IPv6 address]

translate  IN A     [IPv4 address]      ; HTTP: Pootle
           IN AAAA  [IPv6 address]

www        IN CNAME sympa-community.github.io.  ; HTTP: GitHub Pages
```

----
EDIT: SPF record was missing

**Comment by ikedas** [comment]
Hi @everyone,

DNS has been up.  In this weekend TLS certificate and DKIM keys will be set up.  Then we can start moving things over to the new domain.

* As I [proposed](https://github.com/orgs/sympa-community/teams/everyone/discussions/26/comments/4), the new devel list `devel@sympa.community` will be migrated at first, and members of this team will be added to it for the time being.  Subscribers of older lists (`sympa-developers` & `sympa-dev`) will be removed and the posts to them will be forwarded to the new list.
* Migration of the other lists and sites will be done after necessary announcements to be made for the next release.

If there are no objections, above will be the decision about the new devel list.

There may be still time to discuss about the other things (such as the new names of the other lists).

**Comment by ikedas** [comment]
Hi @everyone,

  * Now the web site can switch to a new URL at the touch of a button any time. 
  * DNS has started advertising DKIM keys.
  * Mailing list server started working (I found a few minor bugs): https://lists.sympa.community/
 
In this week, I'll migrate devel list.

**Comment by ikedas** [comment]
## Migrating devel list

Now the old list `sympa-developpers` has been migrated to the new list `devel@sympa.community`, and the memebers of this everyone team has been added to the new list.

  * The following persons have not subscribed to the old list and have not published their e-mail addresses on their GitHub profile, so I sent invitations to the new list to the addresses I found on the commit log. If you accept the invitation, you will be added to the list.

    @elbeho, @t0mab, @e-lie, @dotmobo, @ludovicm67

  * I could not find the addresses for the following persons. Please [contact me](mailto:ikeda@conversion.co.jp) and I will add you to the new list.

    @Poukram35, @ingarv

  * Owners (5 persons) on this team are also the owner of the new list.

**Reply by elbeho** [reply]
Hi.
It's probably too late, but don't add me to any devel list of Sympa.
I won't be of any use.

**Reply by ikedas** [reply]
Hi @elbeho , 
Of course no one can force you to keep subscribing. You can unsubscribe yourself.

**Reply by ikedas** [reply]
@elbeho , in the first place, I don't think you have been subscribed if you haven’t accept the invitation.

**Comment by ikedas** [comment]
## Timezone of list server

I realized my mistake. I had set the server timezone to JST (+0900), and when I ran rebuildarc, the dates of several messages shifted. The original server timezone was CET (+0100 / +0200).

The time zone affects not only the date of the archive, but also the time when the digest will be sent.

For now, I'm going to set it to **UTC** (+0000 without DST). This is the same as the time in Reykjavik.

**Comment by ikedas** [comment]
Now we have new domain
-------------------------------
The lists have been migrated to the new server.  Older lists on listes.renater.fr are no longer usable.

* Currently the most of lists are owned by me alone.  I'd be happy to have someone else take on the owner.

----
I found  a few bugs of Sympa during migration work.  On of them is slightly serious.  I'll repoort then tomorrow.

**Comment by ikedas** [comment]
I had completely lost track of this, but references to some content from the new site still pointed to the old site. Now the relevant content was moved to [assets.sympa.community](https://assets.sympa.community).

I also wanted to move past announcements, but decided against it since many had been already removed.
